test/jest/roktManager.spec.ts

```diff
@@ -1027,4 +1039,120 @@ describe('RoktManager', () => {
+            const promise2 = roktManager['deferredCall']<string>('method2', { data: 2 });
+            const promise3 = roktManager['deferredCall']<string>('method3', { data: 3 });
+            
+            // Verify all promises are created
+            expect(promise1).toBeInstanceOf(Promise);
+            expect(promise2).toBeInstanceOf(Promise);
+            expect(promise3).toBeInstanceOf(Promise);
+            
+            // Verify all messages are queued
+            expect(roktManager['messageQueue'].length).toBe(3);
+            
+            // Extract messageIds
+            const messageId1 = roktManager['messageQueue'][0].messageId!;
+            const messageId2 = roktManager['messageQueue'][1].messageId!;
+            const messageId3 = roktManager['messageQueue'][2].messageId!;
+            
+            // Verify all messageIds are unique
+            expect(messageId1).toBeDefined();
+            expect(messageId2).toBeDefined();
+            expect(messageId3).toBeDefined();
+            expect(messageId1).not.toBe(messageId2);
+            expect(messageId2).not.toBe(messageId3);
+            expect(messageId1).not.toBe(messageId3);
+            
+            // Verify all are tracked in pendingPromises
+            expect(roktManager['pendingPromises'].has(messageId1)).toBe(true);
+            expect(roktManager['pendingPromises'].has(messageId2)).toBe(true);
+            expect(roktManager['pendingPromises'].has(messageId3)).toBe(true);
+        });
+    });
+
+    describe('#completePendingPromise', () => {
+        it('should resolve pending promise with success result', async () => {
+            const promise = roktManager['deferredCall']<string>('testMethod', {});
+            const messageId = roktManager['messageQueue'][0].messageId!;
+
+            // Complete the promise with success result
+            roktManager['completePendingPromise'](messageId, 'success result');
+
+            // Promise should resolve with the result
+            await expect(promise).resolves.toBe('success result');
+
+            // Should clean up the pending promise
+            expect(roktManager['pendingPromises'].has(messageId)).toBe(false);
+        });
+
+        it('should reject pending promise with error', async () => {
+            const promise = roktManager['deferredCall']<string>('testMethod', {});
+            const messageId = roktManager['messageQueue'][0].messageId!;
+            const error = new Error('test error');
+
+            // Complete the promise with error (wrapped in rejected promise)
+            roktManager['completePendingPromise'](messageId, Promise.reject(error));
+
+            // Promise should reject with the error
+            await expect(promise).rejects.toThrow('test error');
+
+            // Should clean up the pending promise
+            expect(roktManager['pendingPromises'].has(messageId)).toBe(false);
+        });
+
+        it('should handle async results correctly', async () => {
```

Comment:
Can this be combined with the first test?  Seems like it's mostly doing the same thing...but adding a few extra checks.

Reply:
i was referring to not the test above, but the first test `should resolve pending promise with success result`